File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200127_165353689/Hero.java
import java.util.ArrayList;

import equipment.EquipmentFactory;

// プレイヤー：勇者
public class Hero extends Player {

	// =======================
	// フィールド変数
	// =======================
	Magic[] healMagic= {new Heal()};
	// =======================
	// コンストラクタ
	// =======================
	public Hero(String name) {
		super(name);
	}

	// =======================
	// Getter / Setter
	// =======================

	// =======================
	// protected メソッド
	// =======================
	/**
	 * 名前(name)からキャラクターに必要なパラメータを生成する
	 */
	@Override
	protected void MakeCharacter() {
		// 戦士のパラメータを名前から生成する
		this.defaultHP =(GetNumber(0, 30) + 10) * 10;
		this.defaultMP = GetNumber(1, 30) + 20;
		this.defaultSTR = GetNumber(2, 70) + 30;
		this.defaultDEF = GetNumber(3, 70) + 30;
		this.defaultLUCK = GetNumber(4, 100) + 1;
		this.defaultAGI = GetNumber(5, 50);
		this.paralyze = false;
		this.poison = false;
		this.active=true;

		// 装備の生成
		this.mySword = EquipmentFactory.DecideSword(GetNumber(6,EquipmentFactory.CountSword()));
		this.myArmor = EquipmentFactory.DecideArmor(GetNumber(7,EquipmentFactory.CountArmor()));
		this.myShield = EquipmentFactory.DecideShield(GetNumber(8,EquipmentFactory.CountShield()));
		this.myJewelry = EquipmentFactory.DecideJewelry(GetNumber(9,EquipmentFactory.CountJewelry()));

		this.hp = defaultHP;
		this.mp = defaultMP;
		this.str = defaultSTR + mySword.EquippedSTR(defaultSTR);
		this.def = defaultDEF + myArmor.EquippedDEF(defaultDEF) +myShield.EquippedDEF(defaultDEF);
		this.luck = defaultLUCK;
		this.agi = defaultAGI + myJewelry.EquippedAGI(defaultAGI);

	}

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================
	/**
	 * {@inheritDoc}<br>
	 * 自身のHPが減っている場合は 回復魔法 を使用し、
	 * 自身のHPが減っていない場合は 対象プレイヤー に攻撃を行う
	 * @param attacker {@inheritDoc}
	 * @param defender {@inheritDoc}
	 */
	@Override
	public void Attack(Player activePlayer,ArrayList<Player> passiveMembers) {

		Player passivePlayer = passiveMembers.get(RandomGenerator.RandomRange(0, passiveMembers.size()));

		// 与えるダメージを求める
		System.out.println(activePlayer.GetName() + "　の　攻撃！");
		int damage = activePlayer.CalcDamage(passivePlayer);

		// 求めたダメージを対象プレイヤーに与える
		System.out.println(passivePlayer.GetName() + "　に　" + damage + "　のダメージ！");
		passivePlayer.Damage(damage);

		passivePlayer.Down();

	}
	/**
	 * {@inheritDoc}<br>
	 */
	@Override
	public boolean Heal(Player activePlayer, ArrayList<Player> passiveMembers) {

		Player passivePlayer = passiveMembers.get(0);
		//使用する魔法を決定する
		Magic UseMagic = this.healMagic[RandomGenerator.RandomRange(0,healMagic.length)];

		if (UseMagic.GetUseMP() > this.mp){

			return false;
		}

		UseMagic.effect(activePlayer, passivePlayer);

		return true;
	}


}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Hero.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Hero : Player
{

	// =======================
	// フィールド変数
	// =======================
	Magic[] healMagic = { new Heal() };
	// =======================
	// コンストラクタ
	// =======================
	public Hero(string name) : base(name)
	{
		this.name = name;
		MakeCharacter();
	}

	// =======================
	// Getter / Setter
	// =======================

	// =======================
	// protected メソッド
	// =======================
	/**
	 * 名前(name)からキャラクターに必要なパラメータを生成する
	 */
	protected override void MakeCharacter()
	{
		// 戦士のパラメータを名前から生成する
		this.defaultHP = (base.GetNumber(0, 30) + 10) * 10;
		this.defaultMP = base.GetNumber(1, 30) + 20;
		this.hp = defaultHP;
		this.mp = defaultMP;

		this.str = base.GetNumber(2, 70) + 30;
		this.def = base.GetNumber(3, 70) + 30;
		this.luck = base.GetNumber(4, 100) + 1;
		this.agi = base.GetNumber(5, 50);
		this.paralyze = false;
		this.poison = false;
		this.active = true;


	}

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================
	/**
	 * {@inheritDoc}<br>
	 * 自身のHPが減っている場合は 回復魔法 を使用し、
	 * 自身のHPが減っていない場合は 対象プレイヤー に攻撃を行う
	 * @param attacker {@inheritDoc}
	 * @param defender {@inheritDoc}
	 */

	public override void Attack(Player activePlayer, List<Player> passiveMembers)
	{
		//攻撃対象の決定
		Player passivePlayer = passiveMembers[0];

		if (passiveMembers.Count > 1)
		{
			passivePlayer = passiveMembers[UnityEngine.Random.Range(0, passiveMembers.Count - 1)];

		}

		// 与えるダメージを求める
		LogText.AddLog(string.Format("{0} の 攻撃！", activePlayer.GetName()));
		int damage = activePlayer.CalcDamage(passivePlayer);

		// 求めたダメージを対象プレイヤーに与える
		LogText.AddLog(string.Format("{0} に {1} のダメージ！", passivePlayer.GetName(), damage));
		passivePlayer.Damage(damage);

		passivePlayer.Down();
	}

	/**
	 * {@inheritDoc}<br>
	 */

	public bool HealHP(Player activePlayer, List<Player> passiveMembers)
	{

		Player passivePlayer = passiveMembers[0];
		//使用する魔法を決定する
		Magic UseMagic = this.healMagic[UnityEngine.Random.Range(0, healMagic.Length - 1)];

		if (UseMagic.GetUseMP() > this.mp)
		{

			return false;
		}

		UseMagic.effect(activePlayer, passivePlayer);

		return true;
	}

}